massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -1129,23 +1136,37 @@ impl ProtocolWorker {
     > {
         massa_trace!("protocol.protocol_worker.note_block_from_node", { "node": source_node_id, "block": block });
 
-        // check header
-        let (block_id, endorsement_ids, _is_header_new) = match self
-            .note_header_from_node(&block.header, source_node_id)
-            .await
-        {
-            Ok(Some(v)) => v,
-            Ok(None) => return Ok(None),
-            Err(err) => return Err(err),
+        let (header, operations, operation_merkle_root, slot) = {
+            let stored_block = self.storage.retrieve_block(&block).unwrap();
+            let stored_block = stored_block.read();
+            (
+                stored_block.block.header.clone(),
+                stored_block.block.operations.clone(),
+                stored_block
+                    .block
+                    .header
+                    .content
+                    .operation_merkle_root
+                    .clone(),
+                stored_block.block.header.content.slot,
+            )
```

Comment:
Note the clones here are I believe mostly a result of being unable to hold a lock across async points, see https://github.com/massalabs/massa/issues/2179, so removing them could be a good follow-up.